internal/errgroup/group_test.go

```diff
@@ -0,0 +1,628 @@
+			if err := tt.checkFunc(egctx); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestGet(t *testing.T) {
+	type test struct {
+		name       string
+		beforeFunc func()
+		checkFunc  func(got Group) error
+	}
+
+	tests := []test{
+		{
+			name: "returns new instance when instance object of global is nil",
+			beforeFunc: func() {
+				clearGlobalObject()
+			},
+			checkFunc: func(got Group) error {
+				if got == nil {
+					return errors.New("group is nil")
+				}
+				return nil
+			},
+		},
+
+		func() test {
+			eg, _ := New(context.Background())
+
+			return test{
+				name: "returns old instance when instance object of global is not nil",
+				beforeFunc: func() {
+					instance = eg
+				},
+				checkFunc: func(got Group) error {
+					if !reflect.DeepEqual(got, eg) {
+						return errors.Errorf("not equals. want: %v, but got: %v", eg, got)
+					}
+					return nil
+				},
+			}
+		}(),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer func() {
+				clearGlobalObject()
+			}()
+
+			if tt.beforeFunc != nil {
+				tt.beforeFunc()
+			}
+
+			if err := tt.checkFunc(Get()); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestLimitation(t *testing.T) {
+	type args struct {
+		limit int
+	}
+
+	type field struct {
+		limitation       chan struct{}
+		enableLimitation atomic.Value
+	}
+
+	type test struct {
+		name      string
+		args      args
+		field     field
+		checkFunc func(eg *group) error
+	}
+
+	tests := []test{
+		{
+			name: "store true when limitation is greater than 0",
+			args: args{
+				limit: 10,
+			},
+			field: field{
+				limitation: make(chan struct{}),
+			},
+			checkFunc: func(eg *group) error {
+				if ok := eg.enableLimitation.Load().(bool); !ok {
+					return errors.Errorf("enableLimitation is wrong. want: %v, got: %v", true, false)
+				}
+				return nil
+			},
+		},
+
+		{
+			name: "store false when limitation is 0",
+			checkFunc: func(eg *group) error {
+				if ok := eg.enableLimitation.Load().(bool); ok {
+					return errors.Errorf("enableLimitation is wrong. want: %v, got: %v", false, true)
+				}
+				return nil
+			},
+		},
+
+		{
+			name: "store false when limitation is less than 0",
+			args: args{
+				limit: -1,
+			},
+			field: field{
+				limitation: make(chan struct{}),
+			},
+			checkFunc: func(eg *group) error {
+				if ok := eg.enableLimitation.Load().(bool); ok {
+					return errors.Errorf("enableLimitation is wrong. want: %v, got: %v", true, false)
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			eg := &group{
+				limitation:       tt.field.limitation,
+				enableLimitation: tt.field.enableLimitation,
+			}
+			eg.Limitation(tt.args.limit)
+
+			if err := tt.checkFunc(eg); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func Test_group_Wait(t *testing.T) {
+	type receiver struct {
+		g Group
+	}
+
+	type test struct {
+		name      string
+		receiver  receiver
```

Comment:
- Could you use `g Group` instead of `receiver` ?

Reply:
If I use it, an error will occur due to waitGroup.
group struct has wg(waitgrou.WaitGroup) that is not a pointer. 
If I create a group using a field struct, the copy of the object will run and wg will not work properly

For example, 
```
type field struct {
  wg waitgrou.WaitGroup
}

for _ tt := range tests  {
    t.Run(tt.name, func(t *testing.T) {
        g := &group {
            wg: tt.wg <-  Copy of object because it is not a pointer
        }
    })
}
```